changelog/improvement.7619.md

```diff
@@ -0,0 +1,4 @@
+Bump TensorFlow version to 2.6.
+
+Users can no longer set `TF_DETERMINISTIC_OPS=1` and use GPU(s) because a `tf.errors.UnimplementedError`
+will be thrown by TensorFlow (read more [here](https://github.com/tensorflow/tensorflow/releases/tag/v2.6.0)).
```

Comment:
Where are we mentioning the expected impact on training speed? I think we should add something to the changelog as well about it.

Reply:
Yeah I think I need to talk to Ty about this. My understanding had been we'd discuss how to communicate the impact as part of [this](https://github.com/RasaHQ/research/issues/220) and also that it would happen before this PR is merged, but now I'm not so sure about the timing. Started a discussion in Slack

Reply:
Adjusted wording to mentioned expected impact as per discussion [here](https://rasa-hq.slack.com/archives/C8LEHMTAM/p1633357754298600)